### PR TITLE
Another round of changes required to compile on ARM

### PR DESCRIPTION
**Pull request by kant2002**
- libunwind change is fix compilation type mismatch.  Present in upstream https://github.com/llvm/llvm-project/blob/d480f968ad8b56d3ee4a6b6df5532d485b0ad01e/libunwind/src/libunwind.cpp#L102
- gcinfodecoder.cpp This one looks like leftover from other times? Would like to be educated. This location assumes that `src\coreclr\inc\regdisp.h` included, but `src\coreclr\nativeaot\Runtime\regdisplay.h` included. I again do not understand how this happens and why this is working for other targets.

I try search in runtime repo, https://github.com/dotnet/runtime/search?q=volatileCurrContextPointers but results very limited and looks like this variables not used a lot, so this again adds to the puzzle.

### CODE REVIEW

**Comment by kant2002** [comment]
This is bare minimum until I make https://github.com/dotnet/runtimelab/pull/1308 work.

File: src/coreclr/vm/gcinfodecoder.cpp [hunk 1]
```diff
@@ -1486,23 +1486,23 @@ OBJECTREF* GcInfoDecoder::GetRegisterSlot(
     _ASSERTE(regNum >= 0 && regNum <= 14);
     _ASSERTE(regNum != 13);  // sp
 
-    DWORD **ppReg;
+    uint32_t **ppReg;
```
Comment (jkotas) [inline src/coreclr/vm/gcinfodecoder.cpp]: This needs `#ifdef FEATURE_REDHAWK` similar to other `GcInfoDecoder::GetRegisterSlot` in this file

Comment (kant2002) [reply src/coreclr/vm/gcinfodecoder.cpp]: How I miss that. Now it make sense!

File: src/coreclr/vm/gcinfodecoder.cpp [hunk 2]
```diff
@@ -1486,6 +1486,11 @@ OBJECTREF* GcInfoDecoder::GetRegisterSlot(
     _ASSERTE(regNum >= 0 && regNum <= 14);
     _ASSERTE(regNum != 13);  // sp
 
+#ifdef FEATURE_REDHAWK
+    PTR_UIntNative* ppReg = &pRD->pR0;
+
```
Comment (jkotas) [inline src/coreclr/vm/gcinfodecoder.cpp]: This needs to special case `regNum == 14`

Comment (kant2002) [reply src/coreclr/vm/gcinfodecoder.cpp]: I have a question. I do not fully understand your remark. But after looking at the source code at least I think I can ask reasonable question.

1. What is `regNum == 14`? If I take `REGDISPLAY` from `src\coreclr\nativeaot\Runtime\regdisplay.h` this is `SP`.
2. In the assert above written that `_ASSERTE(regNum != 13);  // sp`, but 13 index is `pLR`.
3. Why special case? I take a look at `TARGET_ARM64` and they just look at plan structure. I probably miss some concept here.

Comment (kant2002) [reply src/coreclr/vm/gcinfodecoder.cpp]: Looks like I got it. I found this definition and that's from where `regNum` values should be used.
https://github.com/dotnet/runtimelab/blob/dc24a19f55ebe8546bce15db4d208ce2b3053355/src/coreclr/nativeaot/libunwind/include/libunwind.h#L603-L605

Comment (jkotas) [reply src/coreclr/vm/gcinfodecoder.cpp]: Right, the REGDISPLAY structure does not have SP in the usual position, and so the code has to compensate for everything above SP. It does not have much to with libunwind.h.

**Review by jkotas — APPROVED** [review]
Looks good. Thank you!

**Pull request merged**